### PR TITLE
Download diagnostic bundle after failed Si test.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This change not only enables the download of diagnostic bundles agian
but also makes `ci/launch_cluster.sh` runnable from macOS.

### CODE REVIEW

File: ci/launch_cluster.sh [hunk 1]
```diff
@@ -2,34 +2,56 @@
...
-echo "Workspace: ${WORKSPACE}"
 echo "Using: ${TEMPLATE}"
 
-PLATFORM=`uname`
-if [ "$PLATFORM" == 'Darwin' ]; then
-    brew install gettext
-else
-    apt-get update && apt-get install -y -t jessie-backports gettext-base wget
+# Ensure envsubst is available.
+if ! command -v envsubst >/dev/null 2>&1; then
```
Comment (wavesoft) [inline ci/launch_cluster.sh]: I would vote against on installing missing dependencies in this file. I am in favour of checking the environment and failing early, like the `jq` requirement above.

For this purpose you could introduce a `install_deps.sh` file that the CI will run first to check the env **and** install missing deps.

File: ci/launch_cluster.sh [hunk 2]
```diff
@@ -2,34 +2,56 @@
...
+    PLATFORM=$(uname)
+    if [ "$PLATFORM" == 'Darwin' ]; then
+        if [ ! -f /usr/local/opt/gettext/bin/envsubst ]; then
+            echo "Installing gettext to get envsubst."
+            brew install gettext
```
Comment (wavesoft) [inline ci/launch_cluster.sh]: Yeah, don't do this without asking the user first 😄

File: ci/launch_cluster.sh [hunk 3]
```diff
@@ -2,34 +2,56 @@
...
+        apt-get update && apt-get install -y -t jessie-backports gettext-base wget
+    fi
+fi
+
+# Ensure dcos-launch is available.
+mkdir -p "$(pwd)/bin"
+PATH=$PATH:"$(pwd)/bin"
+if ! command -v dcos-launch >/dev/null 2>&1; then
+    echo "dcos-launch was not found. Downloading '$(pwd)/bin' ..."
+    wget 'https://downloads.dcos.io/dcos-test-utils/bin/linux/dcos-launch' \
```
Comment (wavesoft) [inline ci/launch_cluster.sh]: What about the OSX binary? Again, this could go on an `install_deps.sh` file.

File: ci/si_pipeline.sh [hunk 4]
```diff
@@ -70,10 +69,10 @@ case $CLUSTER_LAUNCH_CODE in
...
       (cd tests && make init test)
       SI_CODE=$?
-      # if [ ${SI_CODE} -gt 0 ]; then
-      #  download-diagnostics-bundle
-      # fi
-      ./dcos-launch delete
+      if [ ${SI_CODE} -gt 0 ]; then
+        download-diagnostics-bundle
+      fi
+      ./bin/dcos-launch delete
```
Comment (wavesoft) [inline ci/si_pipeline.sh]: What if `dcos-launch` is already available in your system? 

I would recommend here to just modify the PATH and append the `./bin` subfolder.

**Comment by jeschkies** [comment]
Open fails with know SI test issue.

File: ci/launch_cluster.sh (hunk 2 above)
Comment (jeschkies) [reply ci/launch_cluster.sh]: Yeah. That's kind of wild.

File: ci/si_pipeline.sh (hunk 4 above)
Comment (jeschkies) [reply ci/si_pipeline.sh]: So far this script is only run by the CI and it assumes a specific installation path for dcos-launch.

File: ci/launch_cluster.sh (hunk 1 above)
Comment (jeschkies) [reply ci/launch_cluster.sh]: Done.

**Pull request merged**